uefi-test-runner/src/proto/debug.rs

```diff
@@ -1,15 +1,124 @@
+                            )
+                            .expect_success("Error while registering exception callback");
+                    }
+                    #[cfg(target_arch = "x86_64")]
+                    ProcessorArch::X86_64 => {
+                        info!("Registering exception callback");
+                        debug_support
+                            .register_exception_callback(
+                                0,
+                                Some(exception_callback),
+                                ExceptionType::EXCEPT_X64_DEBUG,
+                            )
+                            .expect_success("Error while registering exception callback");
+                        info!("Deregistering exception callback");
+                        debug_support
+                            .register_exception_callback(0, None, ExceptionType::EXCEPT_X64_DEBUG)
+                            .expect_success("Error while deregistering exception callback");
+                    }
+                    #[cfg(target_arch = "aarch64")]
+                    ProcessorArch::AARCH_64 => {
+                        info!("Registering exception callback");
+                        debug_support
+                            .register_exception_callback(
+                                0,
+                                Some(exception_callback),
+                                ExceptionType::EXCEPT_AARCH64_SERROR,
+                            )
+                            .expect_success("Error while registering exception callback");
+                        info!("Deregistering exception callback");
+                        debug_support
+                            .register_exception_callback(
+                                0,
+                                None,
+                                ExceptionType::EXCEPT_AARCH64_SERROR,
+                            )
+                            .expect_success("Error while deregistering exception callback");
+                    }
+                    // if we reach this, we're running on an arch that `build.py` doesn't support
+                    // TODO: Add match arms as we support testing on more archs
+                    _ => unreachable!(),
+                }
+
+                test_invalidate_instruction_cache(debug_support);
+            }
+        }
     } else {
         warn!("Debug protocol is not supported");
     }
 }
+
+fn test_invalidate_instruction_cache(debug_support: &mut DebugSupport) {
+    info!("Invalidating instruction cache");
+    let mut addr = 0x0;
+    let ptr = &mut addr as *mut _ as *mut c_void;
+
+    unsafe {
+        debug_support
+            .invalidate_instruction_cache(0, ptr, 64)
+            // Should always pass, since the spec says this always returns EFI_SUCCESS
+            .expect_success("Error occured while invalidating instruction cache");
+    }
+}
+
+// FIXME: Maybe turn into a closure?
```

Comment:
I don't think it matters much either way, so IMO these FIXMEs can be removed.

Reply:
Removed.